fmlloader/src/main/java/net/minecraftforge/fml/loading/ImmediateWindowProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.fml.loading;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.IntConsumer;
+import java.util.function.IntSupplier;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+
+public interface ImmediateWindowProvider {
+    String name();
+
+    Runnable initialize(String[] arguments);
+
+    void updateFramebufferSize(IntConsumer width, IntConsumer height);
+
+    long setupMinecraftWindow(final IntSupplier width, final IntSupplier height, final Supplier<String> title, final LongSupplier monitor);
+
+    boolean positionWindow(Optional<Object> monitor, IntConsumer widthSetter, IntConsumer heightSetter, IntConsumer xSetter, IntConsumer ySetter);
+
+    <T> Supplier<T> loadingOverlay(Supplier<?> mc, Supplier<?> ri, Consumer<Optional<Throwable>> ex, boolean fade);
+
+    void updateModuleReads(ModuleLayer layer);
+
+    void periodicTick();
+
+    String getGLVersion();
+}
```

Comment:
Some documentation on this would be very nice, in case somebody else implements it. 
But it is not necessary.

Reply:
Yeah. It's theoretically possible to implement this. Documentation is probably a good idea now.